src/reference/GEMM.cpp

```diff
@@ -15,22 +15,12 @@ namespace {
 template <typename Tin, typename Tout>
 void Run(const Tin *a, const Tin *b, Tout *c, size_t M, size_t N, size_t K,
          ccglib::mma::MemOrder output_mem_order) {
-
-// FIXME: When using 'half' output type on an AMD GPU, the computation cannot be
-// done using 'half' type. This is because the required operators, `__half&
-// operator+(const __half& x)` and `__half& operator*(const __half& x)`, are
-// missing. According to the HIP documentation, they should be available. A bug
-// report has been submitted: https://github.com/ROCm/HIP/issues/3690 but the
-// fix has not yet been upstreamed.
-#if defined(__HIP_PLATFORM_AMD__)
+  // Use the output type as compute type, unless the output is a narrow type
+  // because tensor core multiply-add still happens in the wider type
```

Comment:
Rephrasing it would improve readability:
```suggestion
  // Use float as the compute type when Tout is half or bf16 because tensor core 
  // multiply-add operations execute in float precision. Otherwise, preserve Tout.
```